environments/builder/builder.go

```diff
@@ -0,0 +1,154 @@
+package builder
```

Comment:
please add the copyright header

Reply:
Also, please move this out of the `environment/` dir. A top-level `builder` dir would be fine.